Userland/Libraries/LibWeb/ServiceWorker/Registration.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <andrew@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/HashMap.h>
+#include <LibWeb/ServiceWorker/Registration.h>
+
+namespace Web::ServiceWorker {
+
+struct RegistrationKey {
+    StorageAPI::StorageKey key;
+    ByteString serialized_scope_url;
+
+    constexpr bool operator==(RegistrationKey const&) const = default;
+};
+
+// FIXME: Surely this needs hooks to be cleared and manipulated at the UA level
+//        Does this need to be serialized to disk as well?
+static HashMap<RegistrationKey, Registration> s_registrations;
+
+Registration::Registration(StorageAPI::StorageKey storage_key, URL::URL scope, Bindings::ServiceWorkerUpdateViaCache update_via_cache)
+    : m_storage_key(storage_key)
+    , m_scope_url(scope)
+    , m_update_via_cache_mode(update_via_cache)
+{
+}
+
+// https://w3c.github.io/ServiceWorker/#dfn-service-worker-registration-unregistered
+bool Registration::is_unregistered()
+{
+    // A service worker registration is said to be unregistered if registration map[this service worker registration's (storage key, serialized scope url)] is not this service worker registration.
```

Comment:
Similar here - the spec seems to be saying that it shouldn't be set (which does seem weird/wrong considering it is set without the fragment)

Reply:
Ah - but `get` seems to do this

Reply:
Yeah, they're stored in the registration map without the fragment, but not enough places say to use the fragment for my liking.

Reply:
At least in other places in the spec, whenever the language just says "serialized" it means to include the fragment unless it says "with exclude fragment set to true" (since it is set to false by default). So these cases are either a bug in the spec (seems more likely from my reading), or the code IMO

Reply:
All right. I've added fixmes to say that I suspect that all Registration-related serialization assumes exclude fragment, and that I suspect the schedule_job one wants it too.